src/libraries/System.Security.Cryptography/tests/X509Certificates/DynamicChainTests.cs

```diff
@@ -594,6 +596,102 @@ public static void NameConstraintViolation_InvalidGeneralNames()
             });
         }
 
+        [Fact]
+        public static void NameConstraintViolation_ExcludedTree_Upn()
+        {
+            SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+            builder.AddUserPrincipalName("mona@github.com");
+
+            AsnWriter writer = new(AsnEncodingRules.DER);
+            writer.WriteCharacterString(UniversalTagNumber.UTF8String, "@github.com");
```

Comment:
It's probably more hygenic to use something in a reserved namespace, like `someone@test.example`